internal/fs/fs.go

```diff
@@ -2090,20 +2090,27 @@ func (fs *fileSystem) renameHierarchicalDir(ctx context.Context, oldParent inode
 		return err
 	}
 
+	oldDirName := inode.NewDirName(oldParent.Name(), oldName)
+	newDirName := inode.NewDirName(newParent.Name(), newName)
+
 	// If the call for getBucketDirInode fails it means directory does not exist.
 	newDirInode, err := fs.getBucketDirInode(ctx, newParent, newName)
 	if err == nil {
 		// If the directory exists, then check if it is empty or not.
 		if err = fs.checkDirNotEmpty(newDirInode, newName); err != nil {
 			return err
 		}
+
+		// This refers to an empty destination directory.
+		// The RenameFolder API does not allow renaming to an existing empty directory.
+		// As a workaround, we delete the directory first and then perform rename.
+		newParent.Lock()
+		err = newParent.DeleteChildDir(ctx, newName, false, newDirInode)
```

Comment:
Since we are ignoring this error, should we add _ in place of err variable name and add a comment on ignoring the error, as attempt to rename will be done irrespective of the error.

Reply:
Done